### PR TITLE
Support for site-provided dependencies

### PR DESCRIPTION
**Pull request by ashao**
Includes a new function in the `smart` CLI to aid users whose site admins provide most of the packages. Additionally, if the appropriate environment variables are set, SmartSim will be configured to acknowledge the sitewide package installations.

### CODE REVIEW

File: smartsim/_core/_cli/cli.py
Comment (Spartee) [inline smartsim/_core/_cli/cli.py:71]: I think this was meant to be taken out

File: smartsim/_core/_install/builder.py
Comment (Spartee) [inline smartsim/_core/_install/builder.py:44]: I think you ended up calling this ``SMARTSIM_DEP_INSTALL_PATH``

File: smartsim/_core/config/config.py
Comment (Spartee) [inline smartsim/_core/config/config.py:92]: how about 

```python
dependency_path = os.environ.get('SMARTSIM_DEP_INSTALL_PATH', self.core_path)
```

**Review by Spartee — CHANGES_REQUESTED** [review]
couple comments. Will do another pass after looking through the site install PR

File: smartsim/_core/_cli/cli.py
Comment (ashao) [reply smartsim/_core/_cli/cli.py:71]: Good catch

File: smartsim/_core/_install/builder.py
Comment (ashao) [reply smartsim/_core/_install/builder.py:44]: There's the `SMARTSIM_DEP_INSTALL_PATH` which is set in the builder's environment. On the user side of it, the environment modules set `SMARTSIM_DEP_PATH`.

Comment (ashao) [reply smartsim/_core/_install/builder.py:44]: Oh yeah, I remember why I called them different things. It's because of the way that environment variables set things. If I made it consistent we would incorrectly replace lines like:
```
set          smartsim_dep_path SMARTSIM_DEP_INSTALL_PATH

prepend-path LD_LIBRARY_PATH   $smartsim_dep_path/lib
prepend-path PATH              $smartsim_dep_path/bin

setenv       SMARTSIM_DEP_PATH SMARTSIM_DEP_INSTALL_PATH
```
While I could write some extra scripting to go around that, IMO it seems more reasonable to keep this the way it is. The rationale is that `SMARTSIM_DEP_INSTALL_PATH` is for a specific compiler chain, whereas `SMARTSIM_DEP_PATH` is the generic environment variable which can be changed by an environment module.

**Review by Spartee — APPROVED** [review]
Approving and should be merged once other PR is merged.

**Pull request merged**
